packages/api/src/axelar-query/isomorphic.ts

```diff
@@ -319,4 +345,144 @@ export class AxelarQueryAPIClient extends RestService {
+        await AxelarQueryClient.initOrGetAxelarQueryClient({
+          environment: this.env,
+          axelarRpcUrl: this.axelarRpcUrl,
+        });
+    }
+
+    return this.axelarQueryClient.nexus
+      .Chains({ status: ChainStatus.CHAIN_STATUS_ACTIVATED })
+      .then((resp) => resp.chains);
+  }
+
+  /**
+   * Check if a chain is active.
+   * @param chainId the chain id to check
+   * @returns true if the chain is active, false otherwise
+   */
+  public isChainActive(chainId: EvmChain | string): Promise<boolean> {
+    return this.getActiveChains()
+      .then((chains) => chains.map((chain) => chain.toLowerCase()))
+      .then((chains) => chains.includes(chainId.toLowerCase()));
+  }
+
+  /**
+   * Throw an error if any chain in the list is inactive.
+   * @param chainIds A list of chainIds to check
+   */
+  public async throwIfInactiveChains(chainIds: EvmChain[] | string[]) {
+    const results = await Promise.all(
+      chainIds.map((chainId) => this.isChainActive(chainId))
+    );
+
+    for (let i = 0; i < chainIds.length; i++) {
+      if (!results[i]) {
+        throw new Error(
+          `Chain ${chainIds[i]} is not active. Please check the list of active chains using the getActiveChains() method.`
+        );
+      }
+    }
+  }
+
+  /**
+   * Gets the base fee in native token wei for a given source and destination chain combination
+   * @param sourceChainName
+   * @param destinationChainName
+   * @param sourceTokenSymbol (optional)
+   * @returns base fee in native token in wei, translated into the native gas token of choice
+   */
+  public async getNativeGasBaseFee(
+    sourceChainId: EvmChain,
+    destinationChainId: EvmChain,
+    sourceTokenSymbol?: GasToken,
+    symbol?: string,
+    destinationContractAddress?: string,
+    sourceContractAddress?: string,
+    amount?: number,
+    amountInUnits?: bigint | string
+  ): Promise<BaseFeeResponse> {
+    await throwIfInvalidChainIds([sourceChainId, destinationChainId], this.env);
+    await this.throwIfInactiveChains([sourceChainId, destinationChainId]);
+
+    const params: {
+      method: string;
+      destinationChain: EvmChain;
+      sourceChain: EvmChain;
+      sourceTokenSymbol?: string;
+      symbol?: string;
+      amount?: number;
+      amountInUnits?: bigint | string;
+      destinationContractAddress?: string;
+      sourceContractAddress?: string;
+    } = {
+      method: "getFees",
+      destinationChain: destinationChainId,
+      sourceChain: sourceChainId,
+    };
+    if (sourceTokenSymbol) params.sourceTokenSymbol = sourceTokenSymbol;
+    if (symbol) params.symbol = symbol;
+    if (destinationContractAddress)
+      params.destinationContractAddress = destinationContractAddress;
+    if (sourceContractAddress)
+      params.sourceContractAddress = sourceContractAddress;
+    if (amount) {
+      params.amount = amount;
+    } else if (amountInUnits) {
+      params.amountInUnits = amountInUnits;
+    }
```

Comment:
can you clean this up? does it work if we don't check if the variable is defined before assigning it to params?